cpp/ql/src/Likely Bugs/Memory Management/PointerOverflowCheck-bad.cpp

```diff
@@ -0,0 +1,3 @@
+bool check_pointer_overflow(P *ptr) {
+    return ptr + 0x12345678 < ptr; // BAD
```

Comment:
This is certainly `BAD`, but it's not representative of the code that we've seen flagged by this query. It's not doing anything sensible, even on a compiler like MSVC that doesn't optimize it away -- it's just testing whether `ptr` is near the end of the address space.

The typical examples we see in the results of this query are the same examples shown in the two references in the qhelp: an offset should be bounds-checked as being within the span of two pointers. This looks like

```c
bool not_in_range(T *ptr, T *ptr_end, size_t a) {
    return ptr + a < ptr_end || ptr + a < ptr; // BAD
}
```

It would be helpful to make our `GOOD` example be a transformation of this into something that's valid C:

```c
bool not_in_range(T *ptr, T *ptr_end, size_t a) {
    return a < ptr_end - ptr; // GOOD
}
```

Please add these as qltests too.